### PR TITLE
Fixed #1009: SXFMFormat bug with group cardinality

### LINKED ISSUE #1009: SXFMFormat bug with group cardinality
#### Prerequisitives
* **Eclipse Version:**  2020-03
* **FeatureIDE-Version:**  3.6.2
* **Git branch/Commit-ID:**  develop

### Issue description
Imagine that you have this fm in SXFM format. There is a group cardinality where 3 was used instead of *
```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<feature_model name="FeatureIDE model">
    <feature_tree>
:r FM(FM)
	:g [1,3]
		: A(A)
		: B(B)
		: C(C)
</feature_tree>
    <constraints>
</constraints>
</feature_model>
```

When you export as FeatureIDE xml format  you obtain the result below. First it will be better if no constraint is created as 3 is the number of features in the group (An OR relation will be better). There is some code about this check in SXFMFormat but it seems to fail. Second, the exported FM is not correct as there is an error saying that in the constraint it appears a new FM_1 that does not exist.

```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<featureModel>
	<struct>
		<and mandatory="true" name="FM">
			<feature name="A"/>
			<feature name="B"/>
			<feature name="C"/>
		</and>
	</struct>
	<constraints>
		<rule>
			<conj>
				<disj>
					<not>
						<var>FM_1</var>
					</not>
					<var>A</var>
					<var>B</var>
					<var>C</var>
				</disj>
			</conj>
		</rule>
	</constraints>
</featureModel>
```

### PR DESCRIPTION
**Pull request by Subaro**

### CODE REVIEW

**Comment by Subaro** [comment]
Fixes #1009

File: plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/sxfm/SXFMFormat.java [hunk 1]
```diff
@@ -457,14 +457,17 @@ private void buildFeatureTree(BufferedReader reader) throws UnsupportedModelExce
...
-					} else if ((lineText.contains("[")) && (lineText.contains("]"))) {
-						final int index = lineText.indexOf('[');
-						final int start = Character.getNumericValue(lineText.charAt(index + 1));
-						final int end = Character.getNumericValue(lineText.charAt(index + 3));
+					if ((lineText.contains("[")) && (lineText.contains("]"))) {
+						final String substring = lineText.substring(lineText.indexOf('[') + 1, lineText.indexOf(']'));
+						final String[] numbers = substring.split(",");
+						final int start = Integer.parseInt(numbers[0]);
+						int end = 0;
+						if (numbers[1].equals("*")) {
```
Comment (h3ssto) [inline plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/sxfm/SXFMFormat.java:465]: Are we sure that `numbers` always has two fields?

Comment (Subaro) [reply plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/sxfm/SXFMFormat.java:465]: We only accept a group definition of the following form: `:g [A,B]`. `A` can only be a number while `B` can be a number or the `*`. Other cases are not accepted when parsing. So yes more or less than two number results in an exception that is marked for the user.

Comment (Subaro) [reply plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/sxfm/SXFMFormat.java:465]: See other reply

File: plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/sxfm/SXFMFormat.java [hunk 2]
```diff
@@ -462,9 +462,10 @@ private void buildFeatureTree(BufferedReader reader) throws UnsupportedModelExce
 					} else if (lineText.contains("[1,*]")) {
 						lastFeat.getFeature().getStructure().setOr();
 					} else if ((lineText.contains("[")) && (lineText.contains("]"))) {
-						final int index = lineText.indexOf('[');
-						final int start = Character.getNumericValue(lineText.charAt(index + 1));
-						final int end = Character.getNumericValue(lineText.charAt(index + 3));
+						final String substring = lineText.substring(lineText.indexOf('[') + 1, lineText.indexOf(']'));
+						final String[] numbers = substring.split(",");
+						final int start = Integer.parseInt(numbers[0]);
+						final int end = Integer.parseInt(numbers[1]);
```
[the remark this answers is not in the document]
Comment (Subaro) [reply plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/sxfm/SXFMFormat.java]: > 
> 
> We only accept a group definition of the following form: `:g [A,B]`. `A` can only be a number while `B` can be a number or the `*`. Other cases are not accepted when parsing. So yes, more or less than two number results in an exception that is marked for the user.

**Pull request merged**